.github/workflows/javascript_publish.yml

```diff
@@ -0,0 +1,71 @@
+name: Javascript Publish
+
+on:
+  push:
+    tags:
+      - '*'
+
+jobs:
+  checkTag:
+    runs-on: ubuntu-latest
+    outputs:
+      package: ${{ steps.tagInfo.outputs.package }}
+      version: ${{ steps.tagInfo.outputs.version }}
+    steps:
+      - name: get info from tag
+        id: tagInfo
+        run: |
+          TAG=${GITHUB_REF##*/}
+          VERSION=$(echo "$TAG" | grep -woP "([0-9]+\.[0-9]+\.[0-9]+)")
+          PACKAGE=$(echo "$TAG" | grep -oP "javascript-(.+)" | sed 's/javascript-//')
+          if [ -d "javascript/modules/$PACKAGE" ]; then
+            echo "::set-output name=package::$PACKAGE"
+            echo "::set-output name=version::$VERSION"
+          fi
+  publish:
+    runs-on: ubuntu-latest
+    needs: [checkTag]
+    steps:
+      - name: Validate outputs
+        if: ${{ needs.checkTag.outputs.package }}
+        run: |
+          if [ -z "${{ needs.checkTag.outputs.package }}" ]; then
+            echo "::error ::No package found in tag"
+            exit 1
+          fi
+          if [ -z "${{ needs.checkTag.outputs.version }}" ]; then
+            echo "::error ::No version found in tag"
+            exit 1
+          fi
+      - uses: actions/checkout@v2
+        with:
+          token: ${{ secrets.ARABOT_PAT }}
+      - name: Test
+        uses: ./.github/workflows/javascript_test.yml
+      - name: Publish
+        run: yarn publish --no-git-tag-version --new-version ${{ needs.checkTag.outputs.version }}
+        working-directory: javascript/modules/${{ needs.checkTag.outputs.package }}
+        env:
+          NODE_AUTH_TOKEN: ${{secrets.NPM_TOKEN}}
+      - uses: actions/setup-python@v2
+        with:
+          python-version: '3.x'
+          architecture: 'x64'
+      - name: Get Release Notes
+        id: notes
+        run: python ./.github/scripts/generate_release_notes.py javascript/modules/${{ needs.checkTag.outputs.package }}
+      - uses: "softprops/action-gh-release@v1"
+        if: startsWith(github.ref, 'refs/tags/')
+        with:
+          prerelease: false
+          body_path: ${{ steps.notes.outputs.notes }}
+      - name: Update Changelog
+        run: |
+          git fetch
+          git pull
+          git config --global user.name "Arabot-1"
+          git config --global user.email "arabot-1@users.noreply.github.com"
+          sed -i "s/\[TBD\]/[TBD]  \n  \n\## [1.0.5\] - $(date)/" "javascript/modules/${{ needs.checkTag.outputs.package }}/CHANGELOG.md"
```

Comment:
Is the `1.0.5` literal expected?

Reply:
uhh nop. good spot. I'll change that shortly